ngraph/test/backend/clamp.in.cpp

```diff
@@ -0,0 +1,418 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/engine/test_engines.hpp"
+#include "util/test_case.hpp"
+#include "util/test_control.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+static string s_manifest = "${MANIFEST}";
+using TestEngine = test::ENGINE_CLASS_NAME(${BACKEND_NAME});
+
+namespace
+{
+    template <typename T, test::TestCaseType tct = test::TestCaseType::STATIC>
+    void clamp_test(const element::Type& type,
+                    const PartialShape& dynamic_shape,
+                    const Shape& static_shape,
+                    const std::vector<T>& input,
+                    double min,
+                    double max,
+                    const std::vector<T>& output)
+    {
+        auto data = make_shared<op::Parameter>(type, dynamic_shape);
+        auto clamp = make_shared<op::Clamp>(data, min, max);
+        auto function = make_shared<Function>(clamp, ParameterVector{data});
+
+        auto test_case = test::TestCase<TestEngine, tct>(function);
+        test_case.template add_input<T>(static_shape, input);
+        test_case.template add_expected_output<T>(static_shape, output);
+        return test_case.run();
+    }
+}
+
+NGRAPH_TEST(${BACKEND_NAME}, clamp_integral)
+{
+    Shape in_shape{6};
+    element::Type et = element::i32;
+
+    float min = 0.4; // ceiled to 1
+    float max = 5.6; // floored to 5
```

Comment:
Can you tell me why this behavior is needed? I mean ceiling the `min` while flooring the `max`, why not he other way around? I don't see anything about it in the spec.

Reply:
Please refer to the latest spec updates of this op: [commit](https://github.com/openvinotoolkit/openvino/pull/4663/files#diff-6c3f1389327c3107839f4bfbb489178040f97b664d3f3a02b7d4669f04c5a9d8R50)